specifyweb/frontend/js_src/lib/components/Interactions/PrepDialogRow.tsx

```diff
@@ -64,14 +65,41 @@ export function PrepDialogRow({
             onValueChange={(): void => handleChange(checked ? 0 : available)}
           />
         </td>
-        <td className="justify-end tabular-nums">
-          {syncFieldFormat(
-            getField(schema.models.CollectionObject, 'catalogNumber'),
-            undefined,
-            preparation.catalogNumber
-          )}
-        </td>
-        <td>{preparation.taxon}</td>
+        {typeof preparation.coId === 'number' ? (
+          <td className="justify-end tabular-nums">
+            <Link.NewTab
+              href={`/specify/view/collectionobject/${preparation.coId}`}
```

Comment:
Use `getResourceViewUrl('CollectionObject', preparation.collectionObjectId)` instead.
Reason: it's easy to make a typo in a string or forget a `/`, but when calling getResourceViewUrl everything is safe:
1. TypeScrpipt checks that getResourceViewUrl is a correct name of the function
2. TypeScript checks that 'CollectionObject' is a valid table name
3. TypeScript checks that preparation.collectionObjectId is a number
4. In the future, if we want to change the URL, we just have to change the function and all the places are updated automatically!